src/charm.py

```diff
@@ -206,34 +205,34 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
         """Handle collect unit status event (called after every event)."""
         if not self.model.relations.get("credentials"):
             event.add_status(BlockedStatus("Keystone is not related"))
+            return
 
         if not self._get_keystone_data():
             event.add_status(WaitingStatus("Waiting for credentials from keystone"))
+            return
 
         if not self.model.relations.get("cos-agent"):
             event.add_status(BlockedStatus("Grafana Agent is not related"))
-
-        upstream_snap = get_installed_snap_service(UPSTREAM_SNAP)
+            return
 
         # this is necessary when doing a charm upgrade coming from revision 27
+        upstream_snap = get_installed_snap_service(UPSTREAM_SNAP)
         if upstream_snap.present:
             event.add_status(
                 BlockedStatus(
                     "golang-openstack-exporter detected. Please see: "
                     "https://charmhub.io/openstack-exporter#known-issues"
                 )
             )
+            return
 
         snap_service = get_installed_snap_service(SNAP_NAME)
 
         if not snap_service.present:
-            event.add_status(
-                BlockedStatus("snap service is not installed, please check snap service")
-            )
-        elif not snap_service.is_active():
-            event.add_status(
-                BlockedStatus("snap service is not running, please check snap service")
-            )
+            raise RuntimeError(f"{SNAP_NAME} snap is not installed, but it should be")
+
+        if not snap_service.is_active():
+            raise RuntimeError(f"{SNAP_NAME} snap service is not active, but it should be")
```

Comment:
Ditto

For example: 'Snap service is not active. Please wait for the installation to complete or manually enable the Smartctl snap if the issue persists.'

Reply:
Updated :)